docs/source/06_nodes_and_pipelines/03_modular_pipelines.md

```diff
@@ -4,406 +4,270 @@
-) + pipeline(lunch_pipeline, inputs={"food": "new_name"})
-```
-
-Remember you can pass `Pipeline` objects in the constructor as well, like in the example below. This approach is cleaner and more idiomatic when you are combining multiple modular pipelines together.
+<details>
+<summary>Click here to see a worked example</summary>
 
 ```python
-final_pipeline = Pipeline(
+cook_pipeline = Pipeline(
     [
-        pipeline(cook_pipeline, outputs={"grilled_meat": "new_name"}),
-        pipeline(lunch_pipeline, inputs={"food": "new_name"}),
-        node(...),
-        ...,
+        node(
+            func=defrost, 
+            inputs="frozen_veg", 
+            outputs="veg", 
+            name="defrost_node"
+        ),
+        node(
+            func=grill, 
+            inputs="veg", 
+            outputs="grilled_veg"
+        ),
     ]
 )
-```
-
-The ``sum`` syntax is also supported for connecting pipelines. Similar to the previous example, using the ``sum`` syntax:
 
-```python
-final_pipeline = sum(
+eat_breakfast_pipeline = Pipeline(
     [
-        pipeline(cook_pipeline, outputs={"grilled_meat": "new_name"}),
-        pipeline(lunch_pipeline, inputs={"food": "new_name"}),
-        ...,
+        node(
+            func=eat_breakfast, 
+            inputs="breakfast_food", 
+            outputs=None
+        )
     ]
 )
-
-# equivalent to
-final_pipeline = (
-    pipeline(cook_pipeline, outputs={"grilled_meat": "new_name"})
-    + pipeline(lunch_pipeline, inputs={"food": "new_name"}),
-    +...,
-)
-```
-
-```eval_rst
-.. note::  ``inputs`` should correspond to the pipeline free inputs, while ``outputs`` are either free or intermediary outputs.
-```
-
-
-## How to use a modular pipeline twice
-Consider the example:
-
-```python
-cook_pipeline = Pipeline(
+eat_lunch_pipeline = Pipeline(
     [
-        node(defrost, "frozen_meat", "meat", name="defrost_node"),
-        node(grill, "meat", "grilled_meat"),
+        node(
+            func=eat_lunch, 
+            inputs="lunch_food", 
+            outputs=None
+        )
     ]
 )
 
-eat_breakfast_pipeline = Pipeline([node(eat_breakfast, "breakfast_food", None)])
-eat_lunch_pipeline = Pipeline([node(eat_lunch, "lunch_food", None)])
+cook_pipeline + eat_breakfast_pipeline + eat_lunch_pipeline 
 ```
 
-Now we need to "defrost" two different types of food and input to different pipelines. But we can't use the `cook_pipeline` twice because the internal dataset names will conflict. We might try to call `pipeline()` and map all datasets, but the conflict from the explicitly set `name="defrost_node"` remains.
+If we visualise the snippet above, we see a disjointed pipeline:
+
+* We need to "defrost" two different types of food via different pipelines.
+* We can't use the `cook_pipeline` twice because the internal dataset names will conflict.
+* Mapping all datasets via the `pipeline()`  wrapper will also cause conflicts.
+
+![cook no namespace](../meta/images/cook_no_namespace.png)
```

Comment:
I just realised this image has "Frozen Meat" on it and not "Vegetables" like in the example.

Reply:
Good spot! I made everything vegan when I was writing this :joy:

Reply:
Fixed!